### ISSUE TITLE
FileNotFoundError during bootstrapping to acquire_nix lock

### ISSUE THREAD

**Issue by kezabelle** [linkedin/shiv#93]
Attempting to follow the example given in the readme, using OSX 10.11.6:
```
$ mktmpenv --python=`which python3.6`
$ which python3
/Users/<username>/Virtualenvs/tmp-cf98a678da0c70e/bin/python3

$ python -V
Python 3.6.2

$ python3 -m pip download boto
[...] Saved ./boto-2.49.0-py2.py3-none-any.whl

$ python3 -m pip install shiv
[...]

$ shiv -o boto.pyz --find-links . --no-index boto
[...] Successfully installed boto-2.49.0

$ ./boto.pyz
Traceback (most recent call last):
  File "/Users/<username>/.pyenv/versions/3.6.2/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/<username>/.pyenv/versions/3.6.2/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "boto.pyz/__main__.py", line 3, in <module>
  File "boto.pyz/_bootstrap/__init__.py", line 127, in bootstrap
  File "boto.pyz/_bootstrap/__init__.py", line 79, in extract_site_packages
  File "boto.pyz/_bootstrap/filelock.py", line 71, in __enter__
  File "boto.pyz/_bootstrap/filelock.py", line 39, in acquire_nix
FileNotFoundError: [Errno 2] No such file or directory: '/Users/<username>/.shiv/boto_cab7037d-ea57-4001-9b01-da4b73096d73.lock'

$ ls -al $HOME/.shiv
ls: /Users/<username>/.shiv: No such file or directory

$ mkdir $HOME/.shiv

$ ./boto.pyz
Python 3.6.2 [...]
>>> 
```

It seems like the lock acquisition assumes the directory exists, but I can't see where that's explained in the documentation.

**Comment by yml** [comment]
I am running into this issue. I confirm that the workaround from @kezabelle is also working for me.
This seems to be a regression in `0.0.42`:

```
         "shiv": {
             "hashes": [
-                "sha256:5e8c487dd4cb1c16dae7aacd18fd5b01ff61a3df7d15904e88a791f0bc43bd41",
-                "sha256:9ea73b60b7478904b522a1dd22fbdddd1a31afa33f564ce5ef5eb615b4b1d72a"
+                "sha256:2cce401f1a04fb5ded5eed692375b9b9fe973cbbbe15827eb102323c93f22740",
+                "sha256:c46912a5eea75586b63d7480a9a5e3354940cd9da500f92df33d5997d9ff39e4"
             ],
             "index": "pypi",
-            "version": "==0.0.41"
+            "version": "==0.0.42"
         },
```

**Comment by lorencarvalho** [comment]
Ah, yes, definitely a regression. Thanks for the detailed report @kezabelle!

I'll get this sorted ASAP

**Comment by lorencarvalho** [comment]
@yml @kezabelle fix is pushed in 0.0.43, let me know if you run into any other issues.
thanks again for the report!
